bin/provision-instances

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/python3
+#
+# This script runs a full instance provision over all non-ephemeral instances
+# on OCIM. Running this script requires Django to be initialized - the easiest
+# way to do this is by evaluating the script from within the Django shell by
+# running:
```

Comment:
Yup - as mentioned above, all my comments can be handled in a follow-up iteration. And imho these iterations should be every time we need to redeploy the instances, rather than separate ones, to keep things concrete. The next iteration doesn't need to address all the comments either - again it can be iterative, and probably should be. Maybe just link to this PR in other tickets that will need to redeploy the instances, and to this comment for describing the approach?

CC @bradenmacdonald